app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
Sufia::Engine.routes.url_helpers.method(:download_path).arity -- interesting.  I had trouble tracking down how sufia.download_path was declared.  Am I wrong that [ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path](http://api.rubyonrails.org/classes/ActionDispatch/Routing/PolymorphicRoutes.html) is responsible?  

I'm new to arity so help me out.  Wouldn't ```def download_path(gf, *more)``` and  ```def download_path(gf, options = {})``` both have [arity of -2](http://ruby-doc.org/core-2.2.0/Method.html#method-i-arity)? I guess this implies that we have the case "For methods written in C, returns -1 if the call takes a variable number of arguments."

I'm happy to change it.  Both work for me.  The only place I see a second argument is calls of the form ```download_path(@generic_file, file: 'thumbnail')```